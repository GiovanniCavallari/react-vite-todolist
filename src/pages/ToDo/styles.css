.todo-container {
  width: 50%;
  height: 100vh;
  padding: 32px;
  margin: 0 auto;
}

.todo-container h1 {
  margin: 0 0 48px;
}

.todo-form-container{
  margin: 0 0 48px;
}

.todo-form-container input {
  margin: 0 0 16px;
}
