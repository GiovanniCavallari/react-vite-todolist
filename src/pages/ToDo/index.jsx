import React, { useState } from 'react';
import Button from '../../components/Button';
import Input from '../../components/Input';
import List from '../../components/List';

import './styles.css';

function ToDo() {
  const [value, setValue] = useState('');
  const [items, setItems] = useState([
    {
      text: 'Sample To Do',
      done: false,
    },
  ]);

  const addTodo = () => {
    if (value !== '') {
      const item = {
        text: value,
        done: false,
      };

      const newItems = [...items, item];
      setItems(newItems);
      setValue('');
    }
  };

  const finishItem = (index) => {
    const newItems = [...items];
    newItems[index].done = true;
    setItems(newItems);
  };

  const removeItem = (index) => {
    const newItems = [...items];
    newItems.splice(index, 1);
    setItems(newItems);
  };

  const handleChange = (e) => {
    setValue(e.target.value);
  };

  const handleClick = () => {
    addTodo();
  };

  return (
    <div className="todo-container">
      <h1>To Do List</h1>

      <div className="todo-form-container">
        <Input
          label="Add To Do"
          name="todo"
          value={value}
          placeholder="Type here..."
          onChange={(e) => handleChange(e)}
        />
        <Button onClick={handleClick}>Adicionar</Button>
      </div>

      <List items={items} onFinish={finishItem} onRemove={removeItem} />
    </div>
  );
}

export default ToDo;
