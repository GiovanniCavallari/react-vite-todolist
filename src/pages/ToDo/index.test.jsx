import { fireEvent, render } from '@testing-library/react';
import ToDo from './index';

describe('Test ToDo Page', () => {
  const component = <ToDo />;

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { asFragment } = render(component);
    expect(asFragment).toMatchSnapshot();
  });

  it('should add a new item on to do list', () => {
    const { asFragment, getByPlaceholderText, getByText, getAllByTestId } = render(component);

    const input = getByPlaceholderText('Type here...');
    const button = getByText('Add');

    fireEvent.change(input, {
      target: { value: 'My test To Do' },
    });

    fireEvent.click(button);

    const listItems = getAllByTestId('list-item');

    expect(asFragment).toMatchSnapshot();
    expect(listItems.length).toBe(2);
  });

  it('should finish an item on to do list', () => {
    const { asFragment, getAllByTestId, getByText } = render(component);

    const finish = getAllByTestId('button')[1];
    fireEvent.click(finish);

    const p = getByText('Sample To Do');

    expect(asFragment).toMatchSnapshot();
    expect(p.className).toContain('done');
  });

  it('should remove an item on to do list', () => {
    const { asFragment, getByPlaceholderText, getByText, getAllByTestId } = render(component);

    const input = getByPlaceholderText('Type here...');
    const button = getByText('Add');

    fireEvent.change(input, {
      target: { value: 'My test To Do' },
    });

    fireEvent.click(button);

    const remove = getAllByTestId('button')[2];
    fireEvent.click(remove);

    const listItems = getAllByTestId('list-item');

    expect(asFragment).toMatchSnapshot();
    expect(listItems.length).toBe(1);
  });
});
