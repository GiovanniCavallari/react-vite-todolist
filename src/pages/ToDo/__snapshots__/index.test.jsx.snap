// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ToDo Page should add a new item on to do list 1`] = `[Function]`;

exports[`Test ToDo Page should finish an item on to do list 1`] = `[Function]`;

exports[`Test ToDo Page should remove an item on to do list 1`] = `[Function]`;

exports[`Test ToDo Page should render correctly 1`] = `[Function]`;
